src/chains/ethereum/block/src/block.ts

```diff
@@ -41,13 +52,57 @@ export class Block {
       // TODO: support actual uncle data (needed for forking!)
       // Issue: https://github.com/trufflesuite/ganache/issues/786
       // const uncles = deserialized[2];
-      const totalDifficulty = deserialized[3];
+
+      let totalDifficulty: Buffer;
+      // if there are 7 serialized fields we are after shanghai
+      // as in shanghai we added `withdrawals` to the block data
+      if (deserialized.length === 7) {
+        this._rawWithdrawals = deserialized[3] || []; // added in Shanghai
+        totalDifficulty = deserialized[4];
+        this._rawTransactionMetaData = deserialized[5] || [];
+        this._size = Quantity.toNumber(deserialized[6]);
+      } else {
+        this._rawWithdrawals = null;
+        totalDifficulty = deserialized[3] as any;
+        this._rawTransactionMetaData = (deserialized[4] || []) as any;
+        this._size = Quantity.toNumber(deserialized[5] as any);
+      }
       this.header = makeHeader(this._raw, totalDifficulty);
-      this._rawTransactionMetaData = deserialized[4] || [];
-      this._size = Quantity.toNumber(deserialized[5]);
     }
   }
 
+  /**
+   * Migrates a serialized Block to the latest version. This should only be
+   * called on serialized data from blocks created before v7.8.0.
+   *
+   * This migration updates the `size` value of the block to the correct value
+   * by re-serializing the block for storage in the db.
+   * @param serialized
+   * @returns
+   */
+  static migrate(serialized: Buffer) {
+    const deserialized = decode<GanacheRawBlock>(serialized);
+    const start = deserialized.slice(0, 3) as EthereumRawBlock;
+    start[1] = start[1].map((oldRawTx: any) => {
+      if (oldRawTx.length === 9) {
+        return oldRawTx; // legacy transactions are fine
+      } else {
+        // `type` is always `< 0x27`, so we can yank the first byte from the
```

Comment:
Should this comment be:

```suggestion
        // `type` is always `< 0x7F`, so we can yank the first byte from the
```

??